py/nativeglue.c

```diff
@@ -229,21 +229,13 @@ STATIC bool mp_native_yield_from(mp_obj_t gen, mp_obj_t send_value, mp_obj_t *re
 
 #if MICROPY_PY_BUILTINS_FLOAT
 
-STATIC mp_obj_t mp_obj_new_float_from_f(float f) {
-    return mp_obj_new_float((mp_float_t)f);
-}
+mp_obj_t mp_obj_new_float_from_f(float f);
```

Comment:
Is this standard C99, that declaring the function here will add a definition of the static-inline version defined in the header file?

Reply:
It will only work if there's another source file including that header, but I do think it's standard. Will try to look it up to make sure though.

Reply:
Complicated matter.. Looked around and I did not find a definitive answer. Possibly this should be `extern inline` to force a definition. However nativeglue.c includes runtime.h which includes mpstate.h which includes obj.h so none of this is needed, easy enough :)

Reply:
> so none of this is needed,

You mean none of these declarations are needed, and it's legal to take the pointer (in `mp_fun_table`) of a `static inline` function?

Reply:
Yes I'm like, 99.9% certain that is required by the standard. The .map file also shows this: nativeglue.o gets the 4 functions locally. 

AFAIK inline means 'compiler, you can inline this if you want', static here means 'compiler, I'm putting this in a header so you can clearly see the definition of what you can inline but it's static because I don't want multiple definitions in case you didn't inline it' and taking the address means 'you cannot inline this thing in this translation unit because then it would not have an address'. However comparing address between different translation units would be wrong but that is not the case here.

I find the standard pretty hard to read though and cannot find the exact place where it says this; Wikipedia explicitly says so though :) But if no-one can confirm we could for safety just introduce locally scoped functions here again.

Reply:
Ok, let's just keep it out.  If it breaks on some compiler then it's easy to add them back explicitly.